### ISSUE TITLE
Paddlehub加载训练好的模型，对于同一张图片，每次predict的结果都不同

### ISSUE THREAD

**Issue by MichaelLiu-TJ** [PaddlePaddle/PaddleFormers#1610]
- 版本、环境信息
1）PaddleHub和PaddlePaddle版本：PaddleHub1.8.2，PaddlePaddle1.8.5
2）系统环境：CentOS 7.6.1810， python 3.7.9

使用如下代码，加载训练好的分类模型，对于同一张图片，每次结果都不同。自动下载的预训练模型与训练时是相同的
# coding:utf-8

import base64

import cv2
import numpy as np
import paddlehub as hub
from paddlehub.dataset.base_cv_dataset import BaseCVDataset

module_eff = "efficientnetb7_imagenet"
checkpoint_dir = "./hdstamp_v2"
dataset_dir = "./data"
test_list_file = "test.txt"
label_list_file = "labels.txt"
use_gpu = False

class DemoDataset(BaseCVDataset):
    def __init__(self):
        self.dataset_dir = dataset_dir
        super(DemoDataset, self).__init__(
            base_path=self.dataset_dir,
            test_list_file=test_list_file,
            label_list_file=label_list_file)

def base64_to_nparr(img):
    img_data = base64.b64decode(img)
    nparr = np.fromstring(img_data, np.uint8)
    img_np = cv2.imdecode(nparr, cv2.IMREAD_COLOR)
    return img_np

def module_initialization():
    module = hub.Module(name=module_eff)
    input_dict, output_dict, program = module.context(trainable=False)

    dataset = DemoDataset()

    data_reader = hub.reader.ImageClassificationReader(
        image_width=module.get_expected_image_width(),
        image_height=module.get_expected_image_height(),
        images_mean=module.get_pretrained_images_mean(),
        images_std=module.get_pretrained_images_std(),
        dataset=dataset)

    feature_map = output_dict["feature_map"]
    feed_list = [input_dict["image"].name]

    # Setup RunConfig for PaddleHub Fine-tune API
    hub_config = hub.RunConfig(
        use_data_parallel=False,
        use_cuda=use_gpu,
        checkpoint_dir=checkpoint_dir)
    task = hub.ImageClassifierTask(
        data_reader=data_reader,
        feed_list=feed_list,
        feature=feature_map,
        num_classes=dataset.num_labels,
        config=hub_config)
    return task

def process(img64, task):
    data = []
    img_np = base64_to_nparr(img64)
    data.append(img_np)
    res = task.predict(data=data, return_result=True, accelerate_mode=False)
    return res

**Comment by haoyuying** [comment]
您好，目前您使用的paddle及paddlehub的版本陈旧，推荐使用paddlehub及paddle均为2.1 以上版本。

**Comment by MichaelLiu-TJ** [comment]
> 您好，目前您使用的paddle及paddlehub的版本陈旧，推荐使用paddlehub及paddle均为2.1 以上版本。

那需要重新再训练模型吗？

**Comment by haoyuying** [comment]
分类模型可以使用resnet50vd_imagenet_ssld重新训练

**Comment by croook** [comment]
paddlehub/finetune/task/classifier_task.py 第146行
`label_list = list(self._base_data_reader.label_map.keys())`
直接将key转list，所以结果是随机的
先排个序即可

**Comment by MichaelLiu-TJ** [comment]
> paddlehub/finetune/task/classifier_task.py 第146行 `label_list = list(self._base_data_reader.label_map.keys())` 直接将key转list，所以结果是随机的 先排个序即可

我试一下吧。然后我观察到分类的score每次都不一样，不知道为啥。
